controllers/topology/desired_state.go

```diff
@@ -29,14 +29,17 @@ import (
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
 	"sigs.k8s.io/cluster-api/controllers/external"
 	"sigs.k8s.io/cluster-api/controllers/topology/internal/contract"
+	"sigs.k8s.io/cluster-api/controllers/topology/internal/extensions/patches"
 	tlog "sigs.k8s.io/cluster-api/controllers/topology/internal/log"
 	"sigs.k8s.io/cluster-api/controllers/topology/internal/scope"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
+var patchEngine = patches.NewEngine()
+
```

Comment:
Could this be passed to the reconciler instead?

Reply:
Sure, just to confirm that I got it right:
* add a patchEngine field in the `ClusterReconciler`
* set it to `patches.NewEngine()` in `controllers/topology.SetupWithManager`
?

Reply:
Correct

Reply:
Sounds good. It was a bit awkward to just use a package variable :/

Done